src/Build/BackEnd/Shared/BuildResult.cs

```diff
@@ -209,7 +204,6 @@ internal BuildResult(BuildRequest request, BuildResult existingResults, string[]
             _nodeRequestId = request.NodeRequestId;
             _circularDependency = false;
             _baseOverallResult = true;
-            _buildRequestDataFlags = request.BuildRequestDataFlags;
```

Comment:
Commenting here so we don't forget when this is redone: One small issue is that `_buildRequestDataFlags` is set only in this constructor. There may be code paths that use the other constructors, where we probably want to copy `_buildRequestDataFlags` from the incoming `result` parameter.